lxd/container_lxc.go

```diff
@@ -3224,54 +3226,34 @@ func (c *containerLXC) Update(args containerArgs, userRequested bool) error {
 		c.localConfig["volatile.idmap.base"] = fmt.Sprintf("%v", base)
 	}
 
-	// Apply disk quota changes
-	for _, m := range addDevices {
-		var oldRootfsSize string
-		for _, m := range oldExpandedDevices {
-			if m["type"] == "disk" && m["path"] == "/" {
-				oldRootfsSize = m["size"]
-				break
-			}
-		}
-
-		if m["size"] != oldRootfsSize {
-			size, err := shared.ParseByteSizeString(m["size"])
-			if err != nil {
-				return err
-			}
-
-			err = c.storage.ContainerSetQuota(c, size)
-			if err != nil {
-				return err
-			}
-		}
+	// Check whether the old local devices have a root disk device.
+	oldRootDiskDevice := containerGetRootDiskDevice(oldExpandedDevices)
+	if oldRootDiskDevice == nil || oldRootDiskDevice["pool"] == "" {
+		return fmt.Errorf("No root disk device found.")
```

Comment:
Doesn't that effectively prevent you from ever fixing the problem, should a container be in that state already?